tests/HttpRepository/HttpRepositoryTest.php

```diff
@@ -409,6 +409,78 @@ public function testUpdateResourceWithoutIri(): void
         $httpRepository->update(new PuppyResourceDto(null, 'foo'));
     }
 
+    public function testPartialUpdateResource(): void
+    {
+        /** @var SerializerInterface $serializer */
+        $serializer = static::$container->get(SerializerInterface::class);
+
+        $response = $this->createMock(ResponseInterface::class);
+        $response
+            ->method('getContent')
+            ->willReturn($serializer->serialize(new Puppy(1, 'foo'), 'jsonld'))
+        ;
+
+        $httpClient = $this->createMock(HttpClientInterface::class);
+        $httpClient
+            ->expects(self::once())
+            ->method('request')
+            ->with(
+                'PATCH',
+                '/api/puppies/1?user=me',
+                [
+                    'base_uri' => 'https://localhost',
+                    'headers' => [
+                        'Content-Type' => 'application/merge-patch+json',
+                        'Accept' => 'application/merge-patch+json',
+                    ],
+                    'body' => '{"iri":"\/puppies\/1","super_name":"foo","color":null,"hairs":[]}',
+                ]
+            )
+            ->willReturn($response)
+        ;
+
+        static::$container->set('test.http_client', $httpClient);
+
+        /** @var PuppyHttpRepository $httpRepository */
+        $httpRepository = static::$container->get(PuppyHttpRepository::class);
+
+        $createdPuppyDto = $httpRepository->partialUpdate(new PuppyResourceDto('/puppies/1', 'foo'), ['user' => 'me']);
```

Comment:
```suggestion
        $updatedPuppyDto = $httpRepository->partialUpdate(new PuppyResourceDto('/puppies/1', 'foo'), ['user' => 'me']);
```
Sorry for the previous typo, could you update it as well please?